### PR TITLE
Changes how robotics control console works

### PR DESCRIPTION
**Pull request by Kubisopplay**
## About The Pull Request
Makes remote detonation a feature only for malfuncioning ais, restricts lockdown to IDs with upload access, makes borgs not unlockable by AI if human locks them. Also makes robotics console cost more to print. Kyler said yes

## Why It's Good For The Game
- Removes an easy way to round remove few players without gameplay,
- Requires slightly more effort to curb malfunctioning silicons.
- Makes cyborg factories more viable.

## Changelog

:cl:
balance: Nanotrasen finally realized that installing a powerful explosive in their most popular cyborg model was stupid, so they replaced it with more robust locking system
/:cl:

### CODE REVIEW

**Comment by AlinaStarkova** [comment]
> "Makes remote detonation a feature only for malfuncioning ais"

so subverted AIs cant blow up their killed borgs for extra damage? really? theres no reason to lock this behind malf (malf could blow up borgs with their machine overload power anyway, they dont even NEED this ability) since asimov AIs are not allowed to blow up their borgs unless they are far enough away from humans to not harm them


> "restricts lockdown to IDs with upload access"

im actually cool with this, its so annoying when chuds run in and spam lockdown on the borgs for the lolz (happened to me 3 times in the past 6 months)


> "makes borgs not unlockable by AI if human locks them."

im.... torn on this, on one hand it makes sense, so AI cant just unlock their borgs, but on the other if a human locks them down just to be a dick (yes, it HAS happened where RD or Cap insists the borg is bad and locks them down and then forgets about them or refuses to unlock them for literally 10+ minutes while doing nothing to ensure the borg is bad or not (also to mention,m AIs can ONLY unlock borgs slaved to them, if the borg is bad, the AI itself is bad, its extremely rare for a human to lock a borg for being bad while not going after the AI/cutting cams)

i think, i would be more ok with this last one if a malf AI can ignore that and unlock their borgs regardless ?

**Comment by Kubisopplay** [comment]
I locked it behind malf ais, since they can emag borgs, and emagged borgs can detonate themselves. Ill ask kyler about it, but that would be a pretty small change.
The last part with unlocking... I dont really know. I dont like the shape of this PR, but its in my opinion better then what we have currently, and there is the inevitable argument of console tug of war. All in all, its a small price to cut two most annoying uses of the console, which are locking for lulz, and blowing up cause its easier then putting in a little work to desubvert.

**Comment by MrDoomBringer** [comment]
Borg lockdown should automatically reset after ~5 minutes. Seems more interesting to me to have to go find the borg and kill it or whatever rather than pushing the instant-GBJ-players button then deconning the console.

**Comment by Donpedrito** [comment]
Overall I like this change, but I do worry that it'd be hell for a cyborg to get locked down, then be unable to get unlocked unless one of the 2 (potentially 1 on lowpop) people with upload access unlocks it. Perhaps it could be reduced to require robotics access, so a roboticist (or, on skeleton crew, scientist) could build a console to lock down or unlock borgs.

**Comment by Kubisopplay** [comment]
> Overall I like this change, but I do worry that it'd be hell for a cyborg to get locked down, then be unable to get unlocked unless one of the 2 (potentially 1 on lowpop) people with upload access unlocks it. Perhaps it could be reduced to require robotics access, so a roboticist (or, on skeleton crew, scientist) could build a console to lock down or unlock borgs.
> 
> I'm biased because I often play cyborg on lowpop Bagil, and I have one very particular RD in mind that I'd be extremely worried about being the only one on the station with the capability to lock and unlock me.

Currently its robotics ID locked, and yeah, that is a concern, but there are alternative ways to unlock, like getting a human to cut wires. Also in current situation its more frequent for annoying robo to lock you for some werid reason, then someone lock you for legit reason.

**Comment by Donpedrito** [comment]
> Currently its robotics ID locked, and yeah, that is a concern, but there are alternative ways to unlock, like getting a human to cut wires. Also in current situation its more frequent for annoying robo to lock you for some werid reason, then someone lock you for legit reason.

The PR description says upload access, you should update it to accurately reflect that it's robotics access. Other than that I like the PR.

**Comment by Kubisopplay** [comment]
> > Currently its robotics ID locked, and yeah, that is a concern, but there are alternative ways to unlock, like getting a human to cut wires. Also in current situation its more frequent for annoying robo to lock you for some werid reason, then someone lock you for legit reason.
> 
> The PR description says upload access, you should update it to accurately reflect that it's robotics access. Other than that I like the PR.

by currently i mean before this pr

**Comment by jughu** [comment]
> * Removes an easy way to round remove few players without gameplay,
So true bestieee

> * Requires slightly more effort to curb malfunctioning silicons.
It doesn't take much effort to grind uranium and metal so keep going!!

> * Makes cyborg factories more viable.
You can view cyborgs trough their internal camera trough the security cameras might make cyborgs start with their camera randomly on/off roundstart? and then every factory cyborg will always have it off
>

**Comment by Donpedrito** [comment]
> by currently i mean before this pr

Ah, I was under the impression anyone who reached the console could use it, as I've been detonated by overzealous security officers before; thinking back it might have been a HoS with the spare doing that which is what confused me.

**Comment by thgvr** [comment]
massive malf nerf, looks bad

**Comment by Kubisopplay** [comment]
> massive malf nerf, looks bad

Wait, where is it a nerf

**Comment by optimumtact** [comment]
none of thse changes look good

**Comment by Kubisopplay** [comment]
> none of thse changes look good

It would be better to remove robotics console, but kyler didn't let me do that. But this got the "ok" from him

**Comment by optimumtact** [comment]
ok, but you got a no from me because I don't see any reason why these need to be changed

**Comment by ATH1909** [comment]
I'd give this a 😕, but I really don't like the access requirement change, as it makes the ability to stop a malf or subverted AI heavily dependent on the status of a very small group of people (and their IDs). combined with the other changes from this pr, this also makes it very hard to deal with a powertripping rd who arbitrarily locks down your borgs as an AI.

 and if you want to restrict lockdowns to rds and captains, why is it based on upload access instead of rd office access?

**Comment by Kubisopplay** [comment]
> I'd give this a 😕, but I really don't like the access requirement change, as it makes the ability to stop a malf or subverted AI heavily dependent on the status of a very small group of people (and their IDs). combined with the other changes from this pr, this also makes it very hard to deal with a powertripping rd who arbitrarily locks down your borgs as an AI.
> 
> and if you want to restrict lockdowns to rds and captains, why is it based on upload access instead of rd office access? I forget, do HoPs get upload access?

It's not like there is a myriad other tools to fight malf. I restricted it to upload since that sounded right, hop might get upload access, but that's not really a big deal

**Comment by optimumtact** [comment]
Kyler this one is on you.

**Comment by zxaber** [comment]
I'd leave lockdown (and thus lockdown removal) at the Roboticist level, to at least expand out the list of people that can help a bit.

I'd say a hard no to auto-unlocking. Since humanoids can unlock borgs via wire hacking, you have other tools if an asshat is locking borgs and then removing the console. If you want, put in the user's (ID's) name in the borg's Robotact logs, so that you can track down who is doing it.

File: code/game/machinery/computer/robot.dm [hunk 1]
```diff
@@ -3,7 +3,7 @@
 	desc = "Used to remotely lockdown or detonate linked Cyborgs and Drones."
 	icon_screen = "robot"
 	icon_keyboard = "rd_key"
-	req_access = list(ACCESS_ROBOTICS)
+	req_access = list(ACCESS_AI_UPLOAD)
```
Comment (Iamgoofball) [inline code/game/machinery/computer/robot.dm:6]: fuuuuuck no

Comment (Iamgoofball) [reply code/game/machinery/computer/robot.dm:6]: the rest is fine but this is painful as shit when it comes to bad cyborgs; going to result in a lot more "we killed the borg, re-assemble when it stops being a shitter" when bad cyborg players occur

Comment (Iamgoofball) [reply code/game/machinery/computer/robot.dm:6]: at the very least, make this a new access that only the Roboticists and the RD get?

File: code/game/machinery/computer/robot.dm [hunk 2]
```diff
@@ -3,7 +3,7 @@
 	desc = "Used to remotely lockdown or detonate linked Cyborgs and Drones."
```
Comment (Dorsisdwarf) [inline code/game/machinery/computer/robot.dm:3]: If detonation is being changed to a malf ability, needs to have description updated.

Comment (Kubisopplay) [reply code/game/machinery/computer/robot.dm:3]: True, my bad. Will fix it in a moment

File: code/game/machinery/computer/robot.dm (hunk 1 above)
Comment (jughu) [reply code/game/machinery/computer/robot.dm:6]: > at the very least, make this a new access that only the Roboticists and the RD get?

so robotics access.

Comment (Kubisopplay) [reply code/game/machinery/computer/robot.dm:6]: > the rest is fine but this is painful as shit when it comes to bad cyborgs; going to result in a lot more "we killed the borg, re-assemble when it stops being a shitter" when bad cyborg players occur

you have tipping, so killing is excessive in this situation

Comment (Kylerace) [reply code/game/machinery/computer/robot.dm:6]: @Kubisopplay the more i think about it the less i like restricting the access to upload, but i already told you removing detonate was the only thing i require right now. so fix tests and we can get it tm'd and see if it becomes a problem

**Comment by Burgermancoding** [comment]
I worry that this will lead to many prs that attempt to re-balance and fix certain things around these changes and that cyborgs will ultimately play more like humans as a result

**Comment by Kubisopplay** [comment]
> I worry that this will lead to many prs that attempt to re-balance and fix certain things around these changes and that cyborgs will ultimately play more like humans as a result

I had multiple people say literally the same words in coderbus. My response, partially fueled by Riggles exploration of old tegee builds is :  fuck this, borgs are on downward spiral for long, they haven't changed much for years. If breaking the status quo will end in getting them nerfed it will prepare the grounds for more reworks. If every attempt to change them will be made with "Uh oh, this will end in nerfing them later on", then so it be. Lack of change is worse

File: code/game/machinery/computer/robot.dm [hunk 3]
```diff
@@ -77,24 +77,23 @@
...
 				var/mob/living/silicon/robot/R = locate(params["ref"]) in GLOB.silicon_mobs
 				if(can_control(usr, R) && !..())
-					message_admins(span_notice("[ADMIN_LOOKUPFLW(usr)] [!R.lockcharge ? "locked down" : "released"] [ADMIN_LOOKUPFLW(R)]!"))
-					log_silicon("[key_name(usr)] [!R.lockcharge ? "locked down" : "released"] [key_name(R)]!")
-					log_combat(usr, R, "[!R.lockcharge ? "locked down" : "released"] cyborg")
-					R.SetLockdown(!R.lockcharge)
-					to_chat(R, !R.lockcharge ? span_notice("Your lockdown has been lifted!") : span_alert("You have been locked down!"))
-					if(R.connected_ai)
-						to_chat(R.connected_ai, "[!R.lockcharge ? span_notice("NOTICE - Cyborg lockdown lifted") : span_alert("ALERT - Cyborg lockdown detected")]: <a href='?src=[REF(R.connected_ai)];track=[html_encode(R.name)]'>[R.name]</a><br>")
+					if(isAI(usr)&&R.ailockdown&&R.lockcharge||isAI(usr)&&!R.lockcharge||!isAI(usr))
```
Comment (Kylerace) [inline code/game/machinery/computer/robot.dm:84]: space these out also condense the logic a little bit. 
`isAI(usr) && (R.ailockdown && R.lockcharge || !R.lockcharge) || !isAI(usr)`
is better

File: code/modules/mob/living/silicon/robot/robot_defines.dm [hunk 4]
```diff
@@ -94,7 +94,8 @@
 	var/scrambledcodes = FALSE
 	///Boolean of whether the borg is locked down or not
 	var/lockcharge = FALSE
-
+	//Boolean of whether the borg was locked by its AI or nothing
+	var/ailockdown = FALSE
```
Comment (Kylerace) [inline code/modules/mob/living/silicon/robot/robot_defines.dm:98]: make this snake_case

**Review by Kylerace — CHANGES_REQUESTED** [review]

**Comment by carshalash** [comment]
This has made being a silicon worse and seems to just be a buff to robotocists emagging borgs.

**Comment by TheSilverNuke** [comment]
Necro posting but...

>Requires slightly more effort to curb malfunctioning silicons
>Makes cyborg factories more viable

Both of these are false. Any borgs a malf or subverted AI makes will be completely unable to be used as once unlocked they're out of the round permanently or until killed and repurposed. If anything it's significantly harder to play as a borg. I question the thought process behind this PR.

**Comment by EOBGames** [comment]
> Necro posting but...
> 
> > Requires slightly more effort to curb malfunctioning silicons
> > Makes cyborg factories more viable
> 
> Both of these are false. Any borgs a malf or subverted AI makes will be completely unable to be used as once unlocked they're out of the round permanently or until killed and repurposed. If anything it's significantly harder to play as a borg. I question the thought process behind this PR.

Sir, this PR was merged nearly a full year ago, take it to a new PR.

**Pull request merged**